app/models/service_provider.rb

```diff
@@ -21,11 +21,6 @@ class ServiceProvider < ApplicationRecord
     -> { where.not(push_notification_url: nil).where.not(push_notification_url: '') },
   )
 
-  def self.from_issuer(issuer)
-    return NullServiceProvider.new(issuer: nil) if issuer.blank? || issuer.include?("\x00")
```

Comment:
There's only one failing spec left, and it's the null byte issuer

A few options at this point:
1. keep this method only so we can do a null byte check
2. add null byte checks higher up in the middleware layer

I'm considering (2) and have made a PR for that here: https://github.com/18F/identity-idp/pull/5193